cmd/neofs-node/object.go

```diff
@@ -535,3 +546,52 @@ func (e storageEngine) Lock(locker oid.Address, toLock []oid.ID) error {
 func (e storageEngine) Put(o *objectSDK.Object) error {
 	return engine.Put(e.engine, o)
 }
+
+func cachedHeaderSource(getSvc *getsvc.Service, cacheSize int) headerSource {
+	hs := headerSource{getsvc: getSvc}
+
+	if cacheSize > 0 {
+		hs.cache, _ = lru.New[oid.Address, *objectSDK.Object](cacheSize)
+	}
+
+	return hs
+}
+
+type headerSource struct {
+	getsvc *getsvc.Service
+	cache  *lru.Cache[oid.Address, *objectSDK.Object]
+}
+
+type headerWriter struct {
+	h *objectSDK.Object
+}
+
+func (h *headerWriter) WriteHeader(o *objectSDK.Object) error {
+	h.h = o
+	return nil
+}
+
+func (h headerSource) Head(address oid.Address) (*objectSDK.Object, error) {
+	if h.cache != nil {
```

Comment:
Why would it be `nil`?

Reply:
`cacheSize` _may be_ nil so no caching is done then. hardcoded for now but may be useful in the future

Reply:
we have a config param that explicitly says "no caching, please" (but for morph values mostly)

> cacheSize may be zero

i meant you can change one const and it is not caching anymore, you may make it configurable and just init it with a value you read and zero value is already supported. i agree, it may look strange without a real use-case and config param but since it is already done, why not?